python/lib/communication/dmod/communication/message.py

```diff
@@ -211,83 +151,47 @@ def get_response_to_type(cls) -> Type[AbstractInitRequest]:
-        return serial
-
 
 class InvalidMessage(AbstractInitRequest):
     """
     An implementation of :class:`Message` to model deserialized request messages that are not some other valid message
     type.
     """
 
-    event_type: MessageEventType = MessageEventType.INVALID
+    event_type: ClassVar[MessageEventType] = MessageEventType.INVALID
     """ :class:`MessageEventType`: the type of ``MessageEventType`` for which this message is applicable. """
 
-    @classmethod
-    def factory_init_from_deserialized_json(cls, json_obj: dict):
-        """
-        Factory create a new instance of this type based on a JSON object dictionary deserialized from received JSON.
-
-        Parameters
-        ----------
-        json_obj
-
-        Returns
-        -------
-        A new object of this type instantiated from the deserialize JSON object dictionary, or none if the provided
-        parameter could not be used to instantiated a new object.
-        """
-        try:
-            return cls(content=json_obj['content'])
-        except:
-            return None
-
-    def __init__(self, content: dict):
-        self.content = content
-
-    def to_dict(self) -> dict:
-        return {'content': self.content}
+    content: Dict[str, Any]
 
 
 class InvalidMessageResponse(Response):
 
-    response_to_type = InvalidMessage
+    response_to_type: ClassVar[Type[AbstractInitRequest]] = InvalidMessage
     """ The type of :class:`AbstractInitRequest` for which this type is the response"""
 
-    def __init__(self, data=None):
-        super().__init__(success=False,
-                         reason='Invalid Request Message',
-                         message='Request message was not formatted as any known valid type',
-                         data=data)
+    success = False
+    reason: Literal["Invalid Request message"] = "Invalid Request message"
+    message: Literal["Request message was not formatted as any known valid type"] = "Request message was not formatted as any known valid type"
+    data: Optional[Serializable]
 
+    def __init__(self, data: Optional[Serializable]=None, **kwargs):
+        super().__init__(data=data)
+
+
+class HttpCode(Serializable):
+    http_code: int = Field(ge=100, le=599)
 
 class ErrorResponse(Response):
     """
     A response to inform a client of an error that has occured within a request
     """
-    def __init__(self, message: str, http_code: int = None):
-        if not http_code:
-            http_code = 500
-
-        if not isinstance(http_code, int):
-            try:
-                http_code = int(float(http_code))
-            except:
-                http_code = str(http_code)
-        super().__init__(success=False, reason="Error", message=message, data={"http_code": http_code})
+    success = False
+    reason: Literal["Error"] = "Error"
+    data: HttpCode = Field(default_factory=lambda: HttpCode(http_code=500))
+
+    def __init__(self, message: str, http_code: int = None, **kwargs):
```

Comment:
Thanks for the suggestion, @christophertubbs! I agree, that is more readable. @aaraney #TODO

`kwargs` is just so other fields are acceptable to the constructor, but ignored.

Reply:
This class is kind of weird because we only programmatically create it. We never deserialize into an `ErrorResponse` so it feels a strange. Ill try to remedy that in my revisions.